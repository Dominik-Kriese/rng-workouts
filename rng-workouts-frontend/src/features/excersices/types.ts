export const muscleGroups = <const>[
    'Chest',
    'Shoulder',
    'Biceps',
    'Triceps',
    'Upper Back',
    'Latissimus',
    'Quadriceps',
    'Abs',
    'Hamstrings'
]

export type MuscleGroup = typeof muscleGroups[number]

export type Exercise = {
    name: string;
    muscleGroups: MuscleGroup[];
}
