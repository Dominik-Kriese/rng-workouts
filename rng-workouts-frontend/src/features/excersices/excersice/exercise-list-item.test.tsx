import React from 'react';
import {shallow} from 'enzyme';
import 'jest-styled-components';
import ExerciseListItem from './exercise-list-item';
import {Exercise} from '../types';

describe('Exercise Name', () => {
    it('renders the headline in collapsed mode', () => {
        const exercise: Exercise = {
            name: 'Exercise',
            muscleGroups: ['Abs']
        };
        const name = shallow(<ExerciseListItem exercise={exercise}/>).dive().find('div');
        expect(name).toMatchSnapshot();
    });
});
