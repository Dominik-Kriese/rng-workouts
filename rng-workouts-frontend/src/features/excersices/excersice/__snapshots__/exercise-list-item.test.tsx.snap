// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise Name renders the headline in collapsed mode 1`] = `
.c0 {
  padding: 4px 8px;
  margin-top: 4px;
  background-color: #fff;
}

<div
  className="c0"
>
  <ExerciseName
    name="Exercise"
  />
  <MuscleGroups
    muscleGroups={
      Array [
        "Abs",
      ]
    }
  />
</div>
`;
